### DISCUSSION TITLE
Global custom autoinstall variables

### DISCUSSION THREAD

**Discussion by opoplawski** [cobbler/cobbler#3541] [Q&A]
With cobbler 2.8 I was able to throw arbitrary variables into `settings` and then be able to reference them in kickstart snippets, e.g.:
```
location: boulder
backuphost: backup.nwra.com
ntpserver: ntp.nwra.com
```
kickstart:
```
#if $location != 'boulder' or $type == 'laptop'
#set $authconfig = $authconfig + ' with-mkhomedir'
#end if
```
Now the settings.yaml file is validated (a good thing) but I'd still like to define global autoinstall variables.  Any suggestions?  Should we add a `autoinstall_meta` dict item to settings and define them there?

**Comment by SchoolGuy** [comment]
Yes, I believe that this is the right thing to do. We already have a path in the code for `_resolve` that checks settings so extending that to feed `autoinstall_meta` is possible. Of course we can also take any arbitrary key that we agree on and let the user access this in the templates directly.
